### DISCUSSION TITLE
metaplasmidspades --assembly-graph

### DISCUSSION THREAD

**Discussion by INFINITY1993** [ablab/spades#1010] [Q&A] [answered]
Hi,
I want to recover plasmids by directly using .gfa file that generated during the metaspades assembly, whereas there are three .gfa files (assembly_graph_after_simplification.gfa, assembly_graph_with_scaffolds.gfa, and strain_graph.gfa). Which one should I use?
Thanks!

![1659425024625](https://user-images.githubusercontent.com/63569349/182316428-72925a8b-5f12-4a67-b4a1-d8f7b4a99070.png)

**Comment by asl** [comment, marked as answer]
The answer is available from the `spades.log` file, e.g.:

```
 * Assembled contigs are in <path>/contigs.fasta
 * Assembled scaffolds are in <path>/scaffolds.fasta
 * Paths in the assembly graph corresponding to the contigs are in <path>/contigs.paths
 * Paths in the assembly graph corresponding to the scaffolds are in <path>/scaffolds.paths
 * Assembly graph is in <path>/assembly_graph.fastg
 * Assembly graph in GFA format is in <path>/assembly_graph_with_scaffolds.gfa
```

Same information could be found in the manual: https://cab.spbu.ru/files/release3.15.5/manual.html#spadesoutput

Hope this helps

**Reply by INFINITY1993** [reply]
Thanks for quick reply!
Another question is that I could not find --assembly-graph in spades.py. Please help me with this.

![1659429015424](https://user-images.githubusercontent.com/63569349/182329694-b00273f4-c594-4fb3-ade3-4c1f27bd52cf.png)

**Reply by asl** [reply]
Consider using the latest SPAdes.

**Reply by INFINITY1993** [reply]
OK, Thanks!

**Reply by INFINITY1993** [reply]
Hi, dear asl,
In  the latest version of spades, I did successfully run the metaplasmidspades. But it seems that it would succeed regardless of whether the --assembly-graph was configured or not. Thus, it confused me whether and how the metaplasmidspades takes the spades-generated assembly graph as input.

**Reply by asl** [reply]
Well, the duration should be different. As well as the output :)

**Reply by INFINITY1993** [reply]
OK, Thanks!
